GitCommands/Git/GitRevision.cs

```diff
@@ -86,6 +84,76 @@ public BuildInfo BuildStatus
         public string Subject { get; set; } = "";
         [CanBeNull]
         public string Body { get; set; }
+
+        [CanBeNull]
+        public string BodySummary
+        {
+            get
+            {
+                if (string.IsNullOrWhiteSpace(Body) || !HasMultiLineMessage)
+                {
+                    return Body;
+                }
+
+                return _bodySummary ?? (_bodySummary = BuildSummary(Body.TrimEnd()));
```

Comment:
Is there a good reason to cache this field once computed? My intuition is that this shouldn't be cached as it's unlikely to be reused, and requires a reference on every single git revision in memory. Put another way, the increase in size to `GitRevision` for repos with many commits seems 'worse' overall than the cost of allocating and cleaning up a few temporary strings here and there, especially when you consider that in most cases the body summary is just the body unchanged (if I understand correctly).

Reply:
I imagine that if you needed once to display the tool tip, there is a high probability that you need it another time.

Reply:
I doubt that, but don't have any data to back it up. Again, the overhead of recomputing the summary is trivial, but the cost of adding this field to all revisions is non-trivial.